row.py

```diff
@@ -232,6 +232,83 @@ def get_files_from_index(from_location, task_index, task_count, total_size):
     return file_list
 
 
+def generate_remaining_index(from_location, save_location, job_name):
+    """reads file names from the `from_location` and optionally saves the list to the `save_location` as an index.txt
+    file. Cloud storage buckets must start with `gs://`
+    Args:
+        from_location (str): the directory to read the files from. Prefix GSC buckets with gs://.
+        save_location (str): the directory to save the list of files to. An index.txt file will be created within this
+                             directory
+    Returns:
+        list(str): a list of file names
+    """
+
+    original_files = get_files_from_index(from_location, 0, 1, 89537)
+    original_files = [item.strip() for item in original_files]
+    logging.info("original number of files %i", len(original_files))
+    # print(original_files)
+
+    existing_files = list([])
+
+    logging.info('reading files from "%s"', from_location)
+    if from_location.startswith("gs://"):
+        job_prefix = f"{job_name}/mosaics/"
+        storage_client = google.cloud.storage.Client()
+        iterator = storage_client.list_blobs(from_location[5:], max_results=None, versions=False, prefix=job_prefix)
```

Comment:
**issue**: every time this code runs it incurs a cost. We should be respectful of this and use `generate_index` again to incur that cost once. 

**suggestion**: update `generate_index` to be able to create an index of any folder and pattern. The method would take a prefix and a file name or something. For testing it should be a two step process. Download both indexes and then use the new cli option you created to compare them. When we know it works, we can point at the buckets and only run it once.

Reply:
done - I'll let you look over how I changed this